packages/gatsby-source-filesystem/src/create-file-node.js

```diff
@@ -1,79 +1,72 @@
-const slash = require(`slash`)
-const path = require(`path`)
+const crypto = require(`crypto`)
 const fs = require(`fs-extra`)
 const mime = require(`mime`)
+const path = require(`path`)
+const md5File = require(`md5-file/promise`)
 const prettyBytes = require(`pretty-bytes`)
+const slash = require(`slash`)
 
-const md5File = require(`bluebird`).promisify(require(`md5-file`))
-const crypto = require(`crypto`)
+const createFileNode = async (src, createNodeId, opts = {}) => {
+  let node = {
+    absolutePath: slash(src),
+    children: [],
+    cwd: slash(opts.path || process.cwd()),
+    id: createNodeId(src),
+    parent: null,
+    sourceInstanceName: opts.name,
+  }
+
+  const stats = await fs.stat(node.absolutePath)
+
+  node = {
+    ...node,
+    ...path.parse(node.absolutePath),
+    ...stats,
+    absPath: slash(path.resolve(node.absolutePath)),
+    relativePath: slash(path.relative(node.cwd, node.absolutePath)),
+  }
 
-exports.createFileNode = async (
-  pathToFile,
-  createNodeId,
-  pluginOptions = {}
-) => {
-  const slashed = slash(pathToFile)
-  const parsedSlashed = path.parse(slashed)
-  const slashedFile = {
-    ...parsedSlashed,
-    absolutePath: slashed,
-    // Useful for limiting graphql query with certain parent directory
-    relativeDirectory: path.relative(
-      pluginOptions.path || process.cwd(),
-      parsedSlashed.dir
-    ),
+  node = {
+    ...node,
+    accessTime: node.atime,
+    birthTime: node.birthtime,
+    changeTime: node.ctime,
+    extension: node.ext.slice(1).toLowerCase(),
+    internal: stats.isDirectory()
+      ? {
+          contentDigest: crypto
+            .createHash(`md5`)
+            .update(
+              JSON.stringify({
+                absolutePath: node.absolutePath,
+                stats: stats,
+              })
+            )
+            .digest(`hex`),
+          type: `Directory`,
+        }
+      : {
+          contentDigest: await md5File(node.absolutePath),
+          mediaType: mime.getType(node.ext) || `application/octet-stream`,
+          type: `File`,
+        },
+    modifiedTime: node.mtime,
+    prettySize: prettyBytes(node.size),
+    relativeDirectory: path.relative(node.cwd, node.dir),
   }
 
-  const stats = await fs.stat(slashedFile.absolutePath)
-  let internal
   if (stats.isDirectory()) {
-    const contentDigest = crypto
-      .createHash(`md5`)
-      .update(
-        JSON.stringify({ stats: stats, absolutePath: slashedFile.absolutePath })
-      )
-      .digest(`hex`)
-    internal = {
-      contentDigest,
-      type: `Directory`,
-    }
-  } else {
-    const contentDigest = await md5File(slashedFile.absolutePath)
-    const mediaType = mime.getType(slashedFile.ext)
-    internal = {
-      contentDigest,
-      type: `File`,
-      mediaType: mediaType ? mediaType : `application/octet-stream`,
+    node = {
+      ...node,
+      files___NODE: await fs
+        .readdir(node.absolutePath)
```

Comment:
Will you be able to change this so we don't use `fs.readdir`? We should have all the needed information comming from chockidar directly, so we shouldn't need to run more I/O operation than needed. This also won't honor default default `ignored` parameter used with chockidar.